### ISSUE TITLE
Training the model by GPU quickly eats up my memory 

### ISSUE THREAD

**Issue by Rickgewu** [duggalrahul/AlexNet-Experiments-Keras#3]
When I tried to train the Keras alexnet experiment model using GPU, it quickly eats up my system memory after 6 epoch,  and I tried to reduce the sample number from 1024 to 256 per epoch, it lasted a bit longer time then eats up my system memory again.

I have trained several Alexnet models using caffe in Ubuntu, the system memory usage stabilzes to 20%-30% and never goes burst, because it uses GPU memory instead of system memory, so I think in Keras there must be some other reason using the system memory along with training process, but I canot figure out what reason, please help !

**Comment by duggalrahul** [comment]
@Rickgewu. I don't remember encountering such an issue. Probably, this query would better be suited for the Keras community. 

On a side note, does the system memory finish if you further reduce the samples_per_epoch to (say) 1 ?

**Comment by Rickgewu** [comment]
I tried 1 sample per epoch, first time it can finish all 80 epochs training, but memory still went up nearly crashing, even after training completed, the memory is not freed,  I tried the second time training for 300 epochs, the system memory went burst , so  I think there is a memory leak in the code as below:


# this is the augmentation configuration we will use for training
train_datagen = ImageDataGenerator(
        shear_range=0.2,
        zoom_range=0.2,
        horizontal_flip=True,
        vertical_flip=True)

test_datagen = ImageDataGenerator()
                                  

train_generator = train_datagen.flow_from_directory(
        TRAIN_PATH,  
        batch_size=batch_size,
        shuffle=True,
        target_size=input_size[1:],
        class_mode='categorical')  

test_generator = test_datagen.flow_from_directory(
        TEST_PATH,  
        batch_size=batch_size,
        target_size=input_size[1:],
        shuffle=True,
        class_mode='categorical')

alexnet = get_alexnet(input_size,nb_classes,mean_flag)
    
print (alexnet.summary())


sgd = SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)
alexnet.compile(loss='mse',
              optimizer=sgd,
              metrics=['accuracy'])

history = alexnet.fit_generator(train_generator,
                        samples_per_epoch=1,
                        validation_data=test_generator,
                        nb_val_samples=1,
                        nb_epoch=300,
                        verbose=1)

This seems to be clean codes, not sure why it leaks memory so fast and never release after running?

**Comment by Rickgewu** [comment]
This problem is fixed, GPU was not really working before, now I modified the ~/.theanorc as below, it started to work and no more system memory increae during training:

[global] 
openmp=False 
device = gpu 
floatX = float32 
allow_input_downcast=True 
[blas] 
ldflags= 
[gcc] 
cxxflags=-IE:\Anaconda\MinGW 
[nvcc] 
flags = -LE:\Anaconda\libs 
compiler_bindir = C:\Program Files(x86)\Microsoft Visual Studio 10.0\VC\bin 
fastmath = True

**Comment by yjl7212** [comment]
I have the same problem when training the model by CPU. It shows the memory error after 3 epoch while the system memory is 24GB. The theano is 0.9.0 and the keras is 2.0.8. I have no idea why this happened, please help!

**Comment by duggalrahul** [comment]
@yjl7212 ,

This project was written using Keras 1.2. Try downgrading to version 1.2 via the command pip install keras==1.2 .

**Comment by yjl7212** [comment]
@duggalrahul Thank you very much for your help! I finally solve this problem by install theano==0.8.0 and keras==1.2.0. I hope this will be helpful for others.
